### PR TITLE
Intial work on type propgation on variable equality

### PR DESCRIPTION
**Pull request by sivaramaaa**
* `var1 = var2`

Propagate var2 type to var1 , for now covered only basic case , Will soon cover indirect  variable assignment etc ...

* Also a small fix in afta

### CODE REVIEW

**Comment by sivaramaaa** [comment]
I am trying to fix a few leaks and that one regression , so don't merge it yet !

File: libr/core/anal_tp.c [hunk 1]
```diff
@@ -252,28 +258,26 @@ static void type_match(RCore *core, ut64 addr, char *fcn_name, ut64 faddr, const
...
 					case R_ANAL_OP_TYPE_STORE:
-						R_FREE (regname);
+						res = true;
 						break;
 					}
 				}
 			} else if (var && res && (xaddr != UT64_MAX)) { // Type progation using value
-				char *reg = get_src_regname (core, instr_addr);
-				ut64 ptr = get_addr (trace, reg, j);
+				char tmp[10] = {0};
```
Comment (XVilka) [inline libr/core/anal_tp.c]: Make a constant to share the array size for both `tmp` and `regname`.

File: libr/core/anal_tp.c [hunk 2]
```diff
@@ -389,15 +395,15 @@ R_API void r_core_anal_type_match(RCore *core, RAnalFunction *fcn) {
 				bool sign = false;
 				// Forward propgation of function return type
 				if (!resolved && ret_type && ret_reg) {
-					char *reg = get_src_regname (core, aop.addr);
-					if (reg && !strcmp (reg, ret_reg) && var) {
+					char tmp[10] = {0};
```
Comment (XVilka) [inline libr/core/anal_tp.c]: same here.

File: libr/core/anal_tp.c [hunk 3]
```diff
@@ -413,21 +419,39 @@ R_API void r_core_anal_type_match(RCore *core, RAnalFunction *fcn) {
 					var_retype (anal, var, NULL, "signed", addr, false, true);
 				}
 				// lea rax , str.hello ; mov [local_ch], rax;
-				if (str_flag && (type == R_ANAL_OP_TYPE_MOV)) {
-					const char *q = sdb_fmt ("%d.reg.read", cur_idx);
-					if (sdb_array_contains (trace, q, prev_dest, 0)) {
-						var_retype (anal, var, NULL, "const char *", addr, false, true);
+				if (prop && (type == R_ANAL_OP_TYPE_MOV)) {
+					char *vtype = str_flag ? "const char *": prev_type;
+					char reg[10] = {0};
```
Comment (XVilka) [inline libr/core/anal_tp.c]: and here

**Comment by radare** [comment]
lgtm

**Pull request merged**